src/prefect/backend/flow_run.py

```diff
@@ -728,9 +733,10 @@ def get_all_task_runs(self) -> List["TaskRunView"]:
         task_runs = [TaskRunView._from_task_run_data(data) for data in task_run_data]
         # Add to cache
         for task_run in task_runs:
-            self._cache_task_run_if_finished(task_run)
+            if self._cache_task_run_if_finished(task_run):
+                task_runs.remove(task_run)
 
-        return list(self._cached_task_runs.values())
+        return task_runs + list(self._cached_task_runs.values())
```

Comment:
I would prefer something like this instead of modifying `_cache_task_run_if_finished` just for this purpose. I think it's clearer as well

```python
        new_task_runs = [TaskRunView._from_task_run_data(data) for data in task_run_data]
        
        # Combine the retrieved task runs with the cached ones to create a return value
        # before caching the new ones to avoid returning duplicates
        task_runs = new_task_runs + list(self._cached_task_runs.values())
        
        # Add new runs to cache
        for task_run in new_task_runs:
            self._cache_task_run_if_finished(task_run)

        return task_runs
```
        
Thanks again for working on this!  It's get to see someone get involved in some of the lower-level parts.

Reply:
Michael, I made a commit following your suggestion, dropped the comments as the code is self-explanatory. 

Currently we're blocked on our Prefect instance, so for adding the test instance you can wait until we're unblocked (might be a while) or add it yourself. You'd probably have an easier time mocking the task_info body for the Client.

Also, wondering what you think about the comment I made above regarding how the child flow task results are retrieved vs the parent flow.

Lastly, I haven't been able to successfully map `create_child_flow` (using `.map(params)` or `applied_map()`) Just curious if you know about that scenario and if it's supported directly. If not I'll ask in Slack.

Reply:
That all makes sense.

Mapping should work but I haven't tried it. If you can open an issue with the details / errors that'd be great!

Reply:
We figured mapping out now, it's working!